packages/ember-glimmer/lib/environment.js

```diff
@@ -215,19 +213,14 @@ export default class Environment extends GlimmerEnvironment {
       templates
     } = statement;
 
-    assert(`You attempted to overwrite the built-in helper "${key}" which is not allowed. Please rename the helper.`, !(builtInHelpers[key] && this.owner.hasRegistration(`helper:${key}`)));
+    assert(`You attempted to overwrite the built-in helper "${key}" which is not allowed. Please rename the helper.`, !(this.builtInHelpers[key] && this.owner.hasRegistration(`helper:${key}`)));
 
     if (isSimple && (isInline || isBlock)) {
       // 2. built-in syntax
 
-      if (key === 'component') {
-        return DynamicComponentSyntax.create({ args, templates, symbolTable });
-      } else if (key === 'render') {
-        return new RenderSyntax({ args });
-      } else if (key === 'outlet') {
-        return new OutletSyntax({ args });
-      } else if (key === 'mount') {
-        return new MountSyntax({ args });
+      let buildRefinedSyntax = findSyntaxBuilder(key);
+      if (buildRefinedSyntax) {
+        return buildRefinedSyntax({ args, templates, symbolTable });
```

Comment:
While you are at it, can you make these positional arguments? I started doing it in https://github.com/emberjs/ember.js/pull/14172 so I think we need to pass a common set of args with your approach. `this, args, templates, symbolTable` is probably okay.